cmd/srcd-server/engine/web.go

```diff
@@ -0,0 +1,183 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/sirupsen/logrus"
+	"github.com/src-d/engine-cli/api"
+	"github.com/src-d/engine-cli/docker"
+)
+
+const (
+	gitbaseWebName        = "srcd-cli-gitbase-web"
+	gitbaseWebImage       = "srcd/gitbase-web"
+	gitbaseWebPrivatePort = 80
+
+	bblfshWebName        = "srcd-cli-bblfsh-web"
+	bblfshWebImage       = "bblfsh/web"
+	bblfshWebPrivatePort = 80
+)
+
+// StartBblfshWeb starts a bblfsh web.
+func (s *Server) StartBblfshWeb(
+	ctx context.Context,
+	r *api.StartBblfshWebRequest,
+) (*api.StartBblfshWebResponse, error) {
+	if err := s.startBblfshWeb(ctx, int(r.Port)); err != nil {
+		return nil, err
+	}
+
+	return &api.StartBblfshWebResponse{}, nil
+}
+
+func (s *Server) StopBblfshWeb(
+	ctx context.Context,
+	_ *api.StopBblfshWebRequest,
+) (*api.StopBblfshWebResponse, error) {
+	return &api.StopBblfshWebResponse{}, docker.Kill(bblfshWebName)
+}
+
+func (s *Server) startBblfshWeb(
+	ctx context.Context,
+	port int,
+) error {
+	info, err := docker.Info(bblfshWebName)
+	if err != nil && err != docker.ErrNotFound {
+		return err
+	}
+
+	if info != nil {
+		for _, p := range info.Ports {
+			if int(p.PublicPort) == port {
+				return nil
+			}
+		}
+
+		if err := docker.Kill(bblfshWebName); err != nil {
+			return err
+		}
+	}
+
+	return Run(Component{
+		Name:  bblfshWebName,
+		Start: createBblfshWeb,
+		Options: []docker.ConfigOption{
+			docker.WithPort(port, bblfshWebPrivatePort),
+		},
+		Dependencies: []Component{{
+			Name:  bblfshdName,
+			Start: createBbblfshd,
+		}},
+	})
+}
+
+// StartGitbaseWeb starts a gitbase web.
+func (s *Server) StartGitbaseWeb(
+	ctx context.Context,
+	r *api.StartGitbaseWebRequest,
+) (*api.StartGitbaseWebResponse, error) {
+	if err := s.startGitbaseWeb(ctx, int(r.Port)); err != nil {
+		return nil, err
+	}
+
+	return &api.StartGitbaseWebResponse{}, nil
+}
+
+func (s *Server) StopGitbaseWeb(
+	ctx context.Context,
+	_ *api.StopGitbaseWebRequest,
+) (*api.StopGitbaseWebResponse, error) {
+	return &api.StopGitbaseWebResponse{}, docker.Kill(gitbaseWebName)
+}
+
+func (s *Server) startGitbaseWeb(
+	ctx context.Context,
+	port int,
+) error {
+	info, err := docker.Info(gitbaseWebName)
+	if err != nil && err != docker.ErrNotFound {
+		return err
+	}
+
+	if info != nil {
+		for _, p := range info.Ports {
+			if int(p.PublicPort) == port {
+				return nil
+			}
+		}
+
+		if err := docker.Kill(gitbaseWebName); err != nil {
+			return err
+		}
+	}
+
+	return Run(Component{
+		Name:  gitbaseWebName,
+		Start: createGitbaseWeb,
+		Options: []docker.ConfigOption{
+			docker.WithPort(port, bblfshWebPrivatePort),
```

Comment:
this should be gitbaseWebPrivatePort

Reply:
no no, I mean you used `bblfshWebPrivatePort` instead of `gitbaseWebPrivatePort`
I think that's a mistake, is it?